src/Homie/Uptime.cpp

```diff
@@ -4,15 +4,21 @@ using namespace HomieInternals;
 
 Uptime::Uptime()
 : _seconds(0)
-, _lastTick(0) {
+, _lastTick(0)
+, _rolloverCount(0)   {
 }
 
 void Uptime::update() {
   uint32_t now = millis();
-  _seconds += (now - _lastTick) / 1000UL;
+  
+  if (now < _lastTick) {
+    _rolloverCount++;
+  }
+  
   _lastTick = now;
+  _seconds =  (0xFFFFFFFF / 1000 ) * _rolloverCount + (_lastTick / 1000);
```

Comment:
I'm not sure this is right. I think that you loose 1 milli-second every roll over:

If the limit is 0xFF, the previous tick is 0x01 and the current tick is 0x00, then this mean that 0x100 ticks (seconds = 0x100 ./ tickLSB) have happened. However, this logic would give rolloverCount = 1, lastTick = 0, and seconds = 0xFF / tickLSB.